linkerd/app/outbound/src/lib.rs

```diff
@@ -201,7 +182,8 @@ impl<A: OrigDstAddr> Config<A> {
             // Resolves the target via the control plane and balances requests
             // over all endpoints returned from the destination service.
             const DISCOVER_UPDATE_BUFFER_CAPACITY: usize = 10;
-            let balancer_layer = svc::layers()
+            let balancer_stack = endpoint_stack
+                .serves::<Endpoint>()
```

Comment:
`endpoint_stack` has already verified it serves `Endpoint` typed targets. I'm not sure a redundant check is needed here.